src/test/java/io/candydoc/ddd/extract_ddd_concepts/ExtractDDDConceptsUseCaseTest.java

```diff
@@ -307,6 +309,95 @@ void find_interaction_between_two_different_concepts() throws IOException {
                 .build());
   }
 
+  @Test
+  void find_shared_kernel_inside_given_packages() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.valid_bounded_contexts")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            SharedKernelFound.builder()
+                .name("shared_kernel_one")
+                .description("description of shared kernel")
+                .packageName("io.candydoc.sample.valid_bounded_contexts.shared_kernel")
+                .build());
+  }
+
+  @Test
+  void find_core_concepts_inside_shared_kernel() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.valid_bounded_contexts.shared_kernel")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            CoreConceptFound.builder()
+                .simpleName("name of core concept 1 of shared kernel 1")
+                .description("description of core concept 1 of shared kernel 1")
+                .canonicalName(
+                    "io.candydoc.sample.valid_bounded_contexts.shared_kernel.sub_package.CoreConcept1")
+                .packageName("io.candydoc.sample.valid_bounded_contexts.shared_kernel.sub_package")
+                .boundedContext("io.candydoc.sample.valid_bounded_contexts.shared_kernel")
+                .build());
+  }
+
+  @Test
+  void inner_bounded_context_are_forbidden() throws IOException {
```

Comment:
there are two cases: forbidden in bounded contexts, and forbidden in shared kernels